### ISSUE TITLE
[FEATURE] Complete secrets integration

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmCloud#2411]
### Affected project(s)

- [X] #2497
- [X] examples
- [X] wasmCloud host
- [X] wasmCloud CLI (wash)
- [X] capability providers

### Is your feature request related to a problem? Please describe.

This issue is a larger list of what it means to complete the secrets feature in wasmCloud as specified in #2190.

### Describe the solution you'd like

In order to mark this complete, there are additional tasks to complete past #2344

- [X] #2407 for link secrets support
- [X] Return error strings from the secrets-client crate, cleaning up where necessary
- [X] Update #2395 to include a link secret after #2407, what would be great here additionally is a docker compose for the secrets backend + wasmCloud 1.1 rather than `cargo run`ning the binaries
- [X] #2494
- [X] [https://github.com/wasmCloud/wadm/pull/306](https://github.com/wasmCloud/wadm/pull/306)
- [X] [https://github.com/wasmCloud/wadm/pull/307](https://github.com/wasmCloud/wadm/pull/307)
- [X] Update wasmCloud first party capability providers that require potentially sensitive information to fetch these as secrets first, configuration as a backup with a warning
- [ ] Add documentation on [https://wasmcloud.com](https://wasmcloud.com) about the secrets protocol, example, and best practices.
- [X] Add a CLI for managing secrets in the NATS KV secrets backend

### Describe alternatives you've considered

The following would be nice to have:

- [X] Add a client to the nats-kv-secrets backend for easy putting of secrets and mappings
- [ ] ~~Update the nats-kv-secrets crate to not just be a binary, but to allow for embedding within Rust programs~~
- [ ] Validate secrets and policy server subjects as valid NATS subjects

### Additional context

*No response*

**Comment by brooksmtownsend** [comment]
Since secrets docs are getting addressed in https://github.com/wasmCloud/wasmcloud.com/pull/526, I'm closing out this mega-issue (woo!)
